tests/integration/data_sources_and_expectations/test_misconfigured_expectations.py

```diff
@@ -0,0 +1,184 @@
+import datetime as dt
+
+import pandas as pd
+import pytest
+
+import great_expectations.expectations as gxe
+from tests.integration.conftest import parameterize_batch_for_data_sources
+from tests.integration.data_sources_and_expectations.test_canonical_expectations import (
+    ALL_DATA_SOURCES,
+)
+from tests.integration.test_utils.data_source_config.base import DataSourceTestConfig
+
+PANDAS_DATA_SOURCES: list[DataSourceTestConfig] = [
+    ds for ds in ALL_DATA_SOURCES if ds.label.startswith("pandas")
+]
+SPARK_DATA_SOURCES: list[DataSourceTestConfig] = [
+    ds for ds in ALL_DATA_SOURCES if ds.label.startswith("spark")
+]
+SQL_DATA_SOURCES: list[DataSourceTestConfig] = [
+    ds for ds in ALL_DATA_SOURCES if ds not in PANDAS_DATA_SOURCES and ds not in SPARK_DATA_SOURCES
+]
+
+
+@pytest.mark.unit
+def test_parameterization():
+    # Ensure that all data sources are covered
+    assert len(PANDAS_DATA_SOURCES) == 2
+    assert len(SPARK_DATA_SOURCES) == 1
+    assert len(SQL_DATA_SOURCES) == 7
+    assert len(PANDAS_DATA_SOURCES) + len(SPARK_DATA_SOURCES) + len(SQL_DATA_SOURCES) == len(
+        ALL_DATA_SOURCES
+    )
+
+
+class TestNumericExpectationAgainstStrDataMisconfiguration:
+    # Currently bugs with the following (not raising misconfiguration errors at all):
+    #  - sqlite
+    #  - databricks
+    #  - mysql
+    #  - spark-filesystem-csv
+
+    _DATA = pd.DataFrame({"a": ["b", "c"]})
+    _EXPECTATION = gxe.ExpectColumnStdevToBeBetween(
+        column="a",
+        min_value=0,
+        max_value=1,
+        strict_min=True,
+        strict_max=True,
+    )
+
+    @parameterize_batch_for_data_sources(
+        data_source_configs=PANDAS_DATA_SOURCES,
+        data=_DATA,
+    )
+    def test_pandas(self, batch_for_datasource) -> None:
+        self._test_misconfiguration(
+            batch_for_datasource=batch_for_datasource,
+            exception_message="could not convert string to float",
+        )
+
+    @parameterize_batch_for_data_sources(
+        data_source_configs=list(filter(lambda d: d.label == "big-query", SQL_DATA_SOURCES)),
+        data=_DATA,
+    )
+    def test_bigquery(self, batch_for_datasource) -> None:
+        self._test_misconfiguration(
+            batch_for_datasource=batch_for_datasource,
+            exception_message="No matching signature for operator * for argument types: FLOAT64, STRING",  # noqa: E501
+        )
```

Comment:
I don't love creating a new test for each of these - wish I could parameterize the error message alongside the ds test config but that seems incongruent with our current test architecture?